packages/graphql-generator/src/vendor/@graphql-codegen/core/MAINTENANCE.md

```diff
@@ -25,4 +25,19 @@ how to update to any specific version of `@graphql-codegen/core` so that any nee
 
 ### Step 2 - Remove all Promise/async patterns from the copied code
 
-TBD - These instructions will be written as they are being followed with the next change.
\ No newline at end of file
+For the code forked here most of the complexity of adapting `codegen` to be non-async is captured in type changes surfaced by `@aws-amplify/appsync-modelgen-plugin` as `SyncTypes`. If these base types change when we update the package, our derived types to support Sync execution may need to change as well
+
+These are the steps needed to adapt the `graphql-codegen-core` code to run non-async and adhere to the adapted `SyncTypes`:
```

Comment:
Following these instructions with the current target version will result in the same diff proposed with this change give or take minor details or points where discretion is involved (such as how to handle resolving options).